### PR TITLE
Specific Exceptions: Adapting nixnet interface

### PR DESCRIPTION
**Pull request by felixdivo**
Part of #1046.

Since the *NI-XNET* interface was not released in a stable release yet (see #968), we can freely change the exception handling without deprecation notices.

### CODE REVIEW

**Comment by felixdivo** [comment]
@Mergifyio rebase

**Comment by felixdivo** [comment]
I'd appreciate a review of this change

**Comment by felixdivo** [comment]
I'd appreciate a review of this change

File: can/interfaces/nixnet.py [hunk 1]
```diff
@@ -120,16 +118,18 @@ def __init__(
             self.__session_send.start()
             self.__session_receive.start()
 
-        except errors.XnetError as err:
-            raise NiXNETError(function="__init__", error_message=err.args[0]) from None
+        except errors.XnetError as error:
+            raise CanInitializationError(
+                f"{error.args[0]} ({error.error_type})", error.error_code
+            ) from None
```
Comment (hardbyte) [inline can/interfaces/nixnet.py:124]: `raise from None` is to explicitly remove the earlier exception from the traceback right?

**Review by hardbyte — APPROVED** [review]

File: can/interfaces/nixnet.py (hunk 1 above)
Comment (felixdivo) [reply can/interfaces/nixnet.py:124]: Yes. 😄 But all the information should now be in the new exception. That's also how it was before.

**Comment by felixdivo** [comment]
Thanks Brian!

**Pull request merged**